Scripts/General/ModMercs.lua

```diff
@@ -0,0 +1,732 @@
+    if vars.MercNPCLostList == nil then
+        vars.MercNPCLostList = {}
+    end
+
+    -- Mercs: stuff that needs to be carried with save files
+    if vars.MercSaveDataList == nil then
+        vars.MercSaveDataList = {}
+    end
+
+    if not WasInGame and not WasLoaded then
+
+        -- Fill Savedata list
+        for _, Merc in pairs(MercsDB) do
+            local SaveData = table.copy(SMercSaveData)
+            SaveData.NPC_ID = Merc.NPC_ID
+            table.insert(vars.MercSaveDataList, SaveData) 
+        end
+
+        -- Default avaiable mercs
+        if Game.Debug == true then
+            for _, v in pairs(MercsDB) do
+                Merc_MakeAvailableForHire(v.NPC_ID) -- all of them
+            end
+        else
+            Merc_MakeAvailableForHire(525) -- Warder
+        end
+    end
+end
+
+function events.AfterMonsterAttacked(t, attacker)
+
+    if t == nil then return end
+    if attacker == nil then return end
+
+    -- Prevent mercs turning hostile if party attacks them
+    if t.Attacker.Player ~= nil then
+        if t.Monster.NPC_ID > 0 then
+            if ContainsNumber(MercNPCList, t.Monster.NPC_ID) then
+                t.Monster.Ally      = 9000 + t.Monster.NPC_ID
+                t.Monster.Hostile   = false
+            end
+        end
+    end
+end
+
+function events.MonsterKilled(mon, monIndex, defaultHandler)
+
+    if mon == nil then return end
+    if mon.NPC_ID == 0 then
+        return
+    end
+    if not ContainsNumber(MercNPCList, mon.NPC_ID) then
+        return
+    end
+    
+    evt.Add("NPCs",mon.NPC_ID)
+
+    local Merc                  = Merc_GetByID(mon.NPC_ID)
+    local MercSaveData          = Merc_GetSaveDataByID(mon.NPC_ID)
+    MercSaveData.Release        = false
+
+    if (Merc_IsSpecialTag(Merc, const.Mercenary.Special.Undying)) then
+        Merc_ConsumeCharge(Merc)
+    else
+        MercSaveData.Dead       = true
+        MercSaveData.FightsLeft = 0
+    end
+
+    mon.NPC_ID = 0
+end
+
+function events.AfterLoadMap(WasInGame)
+
+    -- Clear lost mercs
+    if vars then
+        if #vars.MercNPCLostList > 0 then
+            for _, mon in Map.Monsters do
+                if mon.NPC_ID > 0 and mon.Group == 35 then
+                    if ContainsNumber(vars.MercNPCLostList, mon.NPC_ID) then
+                        TableRemoveByValue(vars.MercNPCLostList, mon.NPC_ID)
+                        RemoveMonster(mon)
+                    end
+                end
+            end
+        end
+    end
+
+    -- Reset charges
+    local timeDifference = Game.Time - Game.Map.LastVisitTime
+    if timeDifference > const.Day then
+        Merc_ResetChargesForAllMercs()
+    end
+
+    Timer(
+        function()
+            Merc_ResetChargesForAllMercs()
+        end, 
+        const.Day, const.Hour, false)
+end
+
+-- Data Table
+function ParseMercTables(Table)
+
+    -- Core data
+    local FilePath = "Data/Tables/Mercs.txt"
+	local File	= io.open(FilePath)
```

Comment:
It should be `local File = io.open(FilePath)`

Reply:
Also I suspect that this can throw a runtime exception, maybe its worth handling it?